jsonparser.cpp

```diff
@@ -149,14 +192,18 @@ void JsonParser::parseAndSet(const QByteArray &data, const QString label)
     collator.setCaseSensitivity(Qt::CaseSensitive);
 
     std::sort(dataList.begin(), dataList.end(),
-              [&collator](const JsonData &proj1, const JsonData &proj2)
-         {return collator.compare(proj1.name, proj2.name) > 0;});
+              [&collator](const std::shared_ptr<ProjectData>& proj1, const std::shared_ptr<ProjectData> proj2)
+         {return collator.compare(proj1->name, proj2->name) > 0;});
+
+    for (auto& project : dataList)
+    {
+        auto& images = project->images;
+        std::sort(images.begin(), images.end(), compareVersion);
```

Comment:
@kambala, if this sort is commented out there will be no crash.

Reply:
I believe the reason of the crash here is the behaviour of Qt containers with the for-in loop: it might cause a "detach" and deep copy of container. Read the bottom of [this article](https://doc.qt.io/qt-6/foreach-keyword.html) if you want to know the details.

But with Qt's `foreach` you can't reference the original container because a copy is made.

As a result, I suggest not to modify the original code.

Reply:
the simplest solution I managed to find is to use `qSort` instead of `std::sort`. But I don't understand why it crashes...

Reply:
Hmmm, or I could go a step further and go 100% standard C++. Honestly it’s only a hours work to go the final step of QString to std:string.

Reply:
if possible, try to get rid of the `QList<QVariantMap>` for `images` and use some simple `struct`

Reply:
So I think I found the problem, can't try it out till tomorrow.

If the compare function returns true for two values that are equal then `std:sort` could crash. Explained there in the first answer: https://stackoverflow.com/questions/48972158/crash-in-stdsort-sorting-without-strict-weak-ordering.

Reply:
nice catch! Wasn't aware of that.

Just tested: no crash any more! Changed to `if (versionCmp <= 0)`.

Btw you could simplify the last lines to a single `return versionCmp > 0;`.

Reply:
Yes, I’ll clean it up tomorrow along with addressing the rest of the PR comments.

Reply:
But I think you mean:

`return versionComp < 0;`

Reply:
no. Currently `false` should be returned when `versionCmp <= 0` and `true` otherwise which is equivalent to returning `versionCmp > 0`.

Reply:
The compare function should return `true` if `item1 < item2` == `versionCmp < 0`.

Reply:
aha, it means that you current code sorts in descending order. Then you're right, it should be `versionCmp < 0` to obtain the ascending order.